extensions/bspace/README.md

```diff
@@ -0,0 +1,252 @@
+
+# BehaviorSpace Extension for NetLogo
+
+The BehaviorSpace extension allows you to create and run BehaviorSpace experiments from NetLogo code. You can use the
+extension alone or in combination with the GUI BehaviorSpace tool, and you can run experiments created in the GUI tool
+with the extension.
+
+## Using the BehaviorSpace Extension
+
+To use the BehaviorSpace extension in a model, include the line ```extensions [bspace]``` at the top of the code. If
+you are already using another extension, you can add ```bspace``` to the list inside the square brackets.
+
+Once the extension is loaded, you can use it to run experiments from anywhere in the code. Note that you must call
+```clear-all``` to clear the experiments stored in the code, otherwise they will persist between runs of the model.
+
+## Primitives
+
+### Experiment Management
+
+[`bspace:create-experiment`](#bspacecreate-experiment)
+[`bspace:run-experiment`](#bspacerun-experiment)
+
+### Experiment Modification
+
+[`bspace:set-name`](#bspaceset-name)
+[`bspace:set-pre-experiment-commands`](#bspaceset-pre-experiment-commands)
+[`bspace:set-setup-commands`](#bspaceset-setup-commands)
+[`bspace:set-go-commands`](#bspaceset-go-commands)
+[`bspace:set-post-run-commands`](#bspaceset-post-run-commands)
+[`bspace:set-post-experiment-commands`](#bspaceset-post-experiment-commands)
+[`bspace:set-repetitions`](#bspaceset-repetitions)
+[`bspace:set-sequential-run-order`](#bspaceset-sequential-run-order)
+[`bspace:set-run-metrics-every-step`](#bspaceset-run-metrics-every-step)
+[`bspace:set-run-metrics-condition`](#bspaceset-run-metrics-condition)
+[`bspace:set-time-limit`](#bspaceset-time-limit)
+[`bspace:set-exit-condition`](#bspaceset-exit-condition)
+[`bspace:set-metrics`](#bspaceset-metrics)
+[`bspace:set-variables`](#bspaceset-variables)
+[`bspace:set-thread-count`](#bspaceset-thread-count)
+[`bspace:set-table`](#bspaceset-table)
+[`bspace:set-spreadsheet`](#bspaceset-spreadsheet)
+[`bspace:set-stats`](#bspaceset-stats)
+[`bspace:set-lists`](#bspaceset-list)
+[`bspace:set-update-view`](#bspaceset-update-view)
+[`bspace:set-update-plots`](#bspaceset-update-plots)
+
```

Comment:
documentation needs to be added for

- goto-behaviorspace-documentation
- goto-bspace-extension-documentation
- get-default-parallel-runs
- get-recommended-max-parallel-runs